controllers/metal3.io/host_state_machine_test.go

```diff
@@ -942,7 +942,11 @@ func TestErrorCountIncreasedOnActionFailure(t *testing.T) {
 			result := hsm.ReconcileState(info)
 
 			assert.Equal(t, 1, tt.Host.Status.ErrorCount)
-			assert.Equal(t, "some error", tt.Host.Status.ErrorMessage)
+			if tt.ProvisionerErrorOn == "PowerOff" {
+				assert.Equal(t, clarifySoftPoweroffFailure+"some error", tt.Host.Status.ErrorMessage)
```

Comment:
I suggest you add the expected error to scenarios above rather than calculating it (you can still use "some error" as the default).

Reply:
I understand that adding expected error to scenarios would make the testcases more customizable, and also make the test assertion code section looks clean. However, if we follow this approach, there will be two variables added to the testcase struct: inputError and expectedError. The reason is that when poweroff fails, the original error message will be modified by inserting the clarification message to the head of it. Therefore, the input and expected error messages will be different in this case. However, I feel like adding two more variables can make the test looks a bit more complicated than it should be. HDYT? @dtantsur